src/index.ts

```diff
@@ -138,72 +175,103 @@ export default class MondayConnector extends BaseHttpConnector<
   MondayConnectorEventOptions
 > {
   _sdk: MondaySdk
+  private webhookURL: string
+  private webhookId?: string
+  private webhookLastChangedAt?: number
 
   constructor(app: Reshuffle, options: MondayConnectorConfigOptions, id?: string) {
     super(app, options, id)
+    this.webhookURL = validateBaseURL(options.baseURL) + WEBHOOK_PATH
     this._sdk = mondaySdk({ token: options.token })
+    this.app.registerHTTPDelegate(WEBHOOK_PATH, this)
   }
 
   on(
     options: MondayConnectorEventOptions,
-    handler: ({ req, res, next }: { req: Request; res: Response; next: NextFunction }) => void,
-    eventId: string,
+    handler: (event: MondayEvent) => void,
+    eventId?: string,
   ): EventConfiguration {
-    if (!eventId) {
-      eventId = `Monday${options.path}/${this.id}`
+    const boardId = String(options.boardId)
+    if (!MONDAY_ID_REGEX.test(boardId)) {
+      throw new Error(`Invalid board id: ${options.boardId}`)
+    }
+    if (!EventTypes[options.type]) {
+      throw new Error(`Invalid event type: ${options.type}`)
+    }
+    if (
+      options.type === 'ChangeSpecificColumnValue' &&
+      !MONDAY_ID_REGEX.test(options.columnId || '')
+    ) {
+      throw new Error(`Invalid column id for type ChangeSpecificColumnValue: ${options.columnId}`)
     }
-    const event = new EventConfiguration(eventId, this, options)
-    event.options.path = event.options.path || DEFAULT_WEBHOOK_PATH
-    event.options.deleteWebhookOnExit = event.options.deleteWebhookOnExit || true
-    this.eventConfigurations[event.id] = event
 
+    const event = new EventConfiguration(
+      eventId || `Monday${options.boardId}.${options.type}/${this.id}`,
```

Comment:
I'd use Monday/${options.type}/${options.boardId}/${this.id} as it's easier to read and more consistent with other connectors

Reply:
Fixed